experimental/Schemes/IdealSheaves.jl

```diff
@@ -576,6 +578,168 @@ function primary_decomposition(I::Union{<:MPolyQuoIdeal, <:MPolyQuoLocalizedIdea
   return result
 end
 
+## this should go to src/Rings/mpolyquo-localization.jl
+function minimal_primes(I::Union{<:MPolyQuoIdeal, <:MPolyQuoLocalizedIdeal, <:MPolyLocalizedIdeal})
+  Q = base_ring(I)
+  R = base_ring(Q)
+  decomp = minimal_primes(saturated_ideal(I))
+  result = [ideal(Q, Q.(gens(b))) for b in decomp]
+  return result
+end
+
+
+function minimal_associated_points(I::IdealSheaf)
+  X = scheme(I)
+  OOX = OO(X)
+
+  charts_todo = copy(affine_charts(X))            ## todo-list of charts
+
+  associated_primes_temp = Vector{IdDict{AbsSpec, Ideal}}()  ## already identified components
+                                                  ## may not yet contain all relevant charts. but
+                                                  ## at least one for each identified component
+
+# run through all charts and try to match the components
+  while !iszero(length(charts_todo))
+    U = pop!(charts_todo)
+    !is_one(I(U)) || continue                        ## supp(I) might not meet all components
+    components_here = minimal_primes(I(U))
+
+## run through all primes in MinAss(I(U)) and try to match them with previously found ones
+    for i in 1:length(components_here)
+      matches = match_on_intersections(X,U,components_here[i],associated_primes_temp,false)
+      nmatches = length(matches)
+
+      if nmatches == 0                             ## not found
+        add_dict = IdDict{AbsSpec,Ideal}()         ## create new dict
+        add_dict[U] = components_here[i]           ## and fill it
+        push!(associated_primes_temp, add_dict)
+      elseif nmatches == 1                         ## unique match, update it
+        component_index = matches[1]
+        associated_primes_temp[component_index][U] = components_here[i]
+      else                                                ## more than one match, form union
+        target_comp = pop!(matches)
+        merge!(associated_primes_temp[target_comp], associated_primes_temp[x] for x in matches)
+        for x in matches
+          deleteat!(associated_primes_temp,x)
+        end
```

Comment:
This code will only be correct if either `matches` is sorted in descending order, or else if the indices are adjusted to deal with the removal (i.e., to delete the elements at the initial positions 1, 2, 3,  you'd either need `matches` to be equal to `[3,2,1]`, or equal to `[1,1,1]` (or a few other combinations); while `[1,2,3]` would lead to deletion of the elements which are initially at positions 1, 3, 5.

Or you just do
```suggestion
        deleteat!(associated_primes_temp, matches)
```
assuming `matches` really contain the positions that are to be deleted, sorted any way (so `[1,2,3]`, or `[3,2,1]` or any other permutation).


(This of course applies also to the other very similar call to `deleteat!` in this PR)

Reply:
This also avoids sorting the matches when generating them. Thanks for the very useful hint -- somehow I am still a julia beginner..